### PR TITLE
Bugzilla 1229 Create

### PR DESCRIPTION
**Pull request by mjkoster**
RAML and JSON initial files

### CODE REVIEW

File: create.raml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        href:
+          type: string
+
+      body:
+        application/json:
+          schema: create
+            example: |
+              [
+                {
+                  "rt": "oic.r.switch.binary",
```
Comment (mkettrick) [inline create.raml:42]: Should this be an array?  There may be a case where the created Resource has more than one "rt" value.

File: linkupdate.raml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+          href:
+            type: string
+          rel:
+            type: string
+          rt:
+            type: string
+          ins:
+            type: string
+          location-tag:
+            type: string
```
Comment (mkettrick) [inline linkupdate.raml]: Can you use "anchor" as a query parameter?

File: linkupdate.raml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+          application/json:
+            schema: linklist
+              example: |
+                [
+                  {
+                    "href": "ocf://5c68f373-0c4c-4604-9922-db639e20533d/switch",
+                    "rel": ["boundto"],
+                    "bind": "obs",
+                    "anchor": "modeswitch",
+                    "if": ["oic.if.s"],
```
Comment (mkettrick) [inline linkupdate.raml]: Add "oic.if.baseline" to the example.

File: delete.raml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        description: |
+          Delete method processing
+        responses:
+          200:
+            body:
+              application/json:
+                schema: delete-response
+                  example: |
+                    {
+                      "href": "/rules/heat-on/fanmode"
```
Comment (mkettrick) [inline delete.raml]: The "href" is supposed to be relative to the parent Collection, right?  I assume that "/rules/heat-on" is the URI of the Collection.  Or maybe the Collection is "/rules".  May want to change this "href" to make it clear that it's relative to the Collection.

File: linkupdate.raml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+                    "ins": "377fac40"
+                  }
+                ]
+
+          application/merge-patch+json:
```
Comment (mkettrick) [inline linkupdate.raml]: This isn't the patch defined by CoAP, is it?  I'm not sure what "merge-patch+json" means.

File: schemas/oic.if.delete-response-schema.json [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+        "type": "object",
+        "properties": {
+          "href": {
+            "type": "string",
+            "maxLength": 256,
+            "description": "The returned URI is relative to the device root path, i.e. starts with / ",
+            "format": "uri"
+          }
+        },
+        "additionalProperties": true,
```
Comment (mkettrick) [inline schemas/oic.if.delete-response-schema.json:17]: Is it possible to include additional Properties?  I thought only "href" was allowed.  What would it mean if other Properties were included in the DELETE response?

File: schemas/oic.if.linkupdate-schema.json [hunk 7]
```diff
@@ -0,0 +1,72 @@
+{
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "description" : "Copyright (c) 2018 Open Connectivity Foundation, Inc. All rights reserved.",
+    "id": "http://openconnectivityfoundation.github.io/core/schemas/oic.if.linkupdate-schema.json#",
+    "title": "Linkupdate Format",
+    "definitions": {
+      "linkupdate-schema": {
+        "type": ["object", "array"],
+        "properties": {
```
Comment (mkettrick) [inline schemas/oic.if.linkupdate-schema.json]: Should "rt" and "if" be included here?  They are included in the examples in the CR for Bug 1229.

**Comment by rabardini** [comment]
Out of date and no longer needed.

**Pull request closed without merging**